### PR TITLE
Bugfix/weathercleanup

### PR DESCRIPTION
**Pull request by Bensge**
Fix bug in weather plugin and improve code style

### CODE REVIEW

**Comment by Legion2** [comment]
can you replace the use of System.out and System.err with a logger

**Comment by Bensge** [comment]
The only console logs in this plugin are inside the soon to-be-replaced `ConfigLoader.java`

File: plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/ConfigLoader.java [hunk 1]
```diff
@@ -23,22 +23,32 @@
...
-    public ConfigLoader() {
+    @PostConstruct
+    public void init() {
         p = new Properties();
         try {
             p.load(new FileReader("apikeys/weather_config.properties"));
         } catch (IOException e) {
-            System.err.println("Error loading config file for weather plugin");
+            logger.error("Error loading config file for weather plugin");
             e.printStackTrace();
```
Comment (Legion2) [inline plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/ConfigLoader.java]: use `logger.error(message, error)`

File: plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/ConfigLoader.java [hunk 2]
```diff
@@ -23,22 +23,32 @@
...
 import java.io.FileReader;
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Properties;
 
+@Service
 public class ConfigLoader {
 
+    @Reference
+    public Logger logger;
```
Comment (Legion2) [inline plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/ConfigLoader.java]: make the logger private

File: plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/ConfigLoader.java (hunk 1 above)
Comment (Bensge) [reply plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/ConfigLoader.java]: So much work for a class that'll be thrown away soon :/

Comment (Legion2) [reply plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/ConfigLoader.java]: i now but you have to learn how the logger works, so please remove `e.printStackTrace();`

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java [hunk 3]
```diff
@@ -44,54 +50,101 @@
...
 	 */
 	@Test
-	void testGetWeatherReportTomorrow() {
-		// TODO
+	void testGetWeatherTomorrow() {
+		Mockito.when(this.logic.getReportTomorrow()).thenReturn(this.day);
+		
+		Response response = this.target.path("weather").path("tomorrow").request().get();
+		
+		assertEquals(200, response.getStatus());
```
Comment (Legion2) [inline plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: use `assertThat(actuale, matcher)` like `assertThat(response.getStatus(), is(200))`

**Comment by Bensge** [comment]
Rebased on current `dev`. Let's get this merged

**Pull request merged**